pkg/feature/segmentpersister/persister.go

```diff
@@ -338,6 +338,24 @@ func (p *Persister) handleEvent(ctx context.Context, event *serviceevent.BulkSeg
 	}
 	cnt, err := p.persistSegmentUsers(ctx, event.EnvironmentNamespace, event.SegmentId, event.Data, event.State)
 	if err != nil {
+		if err := p.updateSegmentStatus(
```

Comment:
When an error occurs, we must update the segment status. Otherwise, it will stay in the `UPLOADING` status forever.